### PR TITLE
PaymentApplied model added

### PR DESCRIPTION
**Pull request by AshishTI23**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +52,7 @@ The gem is available as open source under the terms of the [MIT License](https:/
 * Lockstep::MagicLink
 * Lockstep::Note
 * Lockstep::Payments
+* Lockstep::PaymentApplied
```
Comment (rishirajkumar97) [inline README.md:55]: can you add the other model added as well ServiceFabricPayment

File: app/models/lockstep/payment_applied.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class Lockstep::PaymentApplied < Lockstep::ApiRecord
+  self.model_name_uri = "v1/payments-applied"
+  self.id_ref = "payment_applied_id"
+  self.query_path = ""
+  load_schema(Schema::PaymentApplied)
+
+  def self.merge_all_attributes(objects, response)
```
Comment (rishirajkumar97) [inline app/models/lockstep/payment_applied.rb:7]: why is this method coded here? can you let me know if this has any difference apart from the general methods which are already in ApiRecord.rb

File: app/models/lockstep/payment_applied.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    objects.each_with_index do |item, index|
+      next unless response[index]
+
+      new_attributes = response[index].transform_keys { |key| key.underscore }
```
Comment (rishirajkumar97) [inline app/models/lockstep/payment_applied.rb:11]: The gem should ideally transform the keys automatically I think?

File: app/models/lockstep/service_fabric_payment.rb [hunk 4]
```diff
@@ -3,5 +3,34 @@ class Lockstep::ServiceFabricPayment < Lockstep::ApiRecord
...
   self.query_path=""
   load_schema(Schema::SfPaymentRequest)
+  
+  # Returning alwas false becase if id_ref is present in
+  # request body it returns true but in this model id_ref
+  # is required field in request body
+  # TODO: Make it generic 
+  def persisted?
+    false
+  end
```
Comment (rishirajkumar97) [inline app/models/lockstep/service_fabric_payment.rb:13]: can you explain why we are using persisted? always as false? so even for an existing object .persisted? will give false which is wrong...

File: app/models/lockstep/service_fabric_payment.rb [hunk 5]
```diff
@@ -3,5 +3,34 @@ class Lockstep::ServiceFabricPayment < Lockstep::ApiRecord
...
+
+  # Preventing deletion of id_ref from puts_attrs because
+  # id_ref column is required to send in body in case of
+  # create ServiceFabricPayment
+  # TODO: Make it generic
+  def attributes_for_saving
```
Comment (rishirajkumar97) [inline app/models/lockstep/service_fabric_payment.rb:19]: this is the same thing in the apirecord.rb class right , how is this method modified and can you let us know the scenario for this override as well

File: README.md (hunk 1 above)
Comment (AshishTI23) [reply README.md:55]: It is already there Lockstep::ServiceFabricPayment

File: app/models/lockstep/payment_applied.rb (hunk 2 above)
Comment (AshishTI23) [reply app/models/lockstep/payment_applied.rb:7]: in create method of `ApiRecord.rb` body was being sent inside an array but for this call we don't need that, Platform will raise exception if we do so

File: app/models/lockstep/service_fabric_payment.rb (hunk 5 above)
Comment (AshishTI23) [reply app/models/lockstep/service_fabric_payment.rb:19]: Here I removed `put_attrs.delete(id_ref)` to not delete `id_ref` key from request body

File: app/models/lockstep/service_fabric_payment.rb (hunk 4 above)
Comment (AshishTI23) [reply app/models/lockstep/service_fabric_payment.rb:13]: Please suggest if there is better way

Comment (rishirajkumar97) [reply app/models/lockstep/service_fabric_payment.rb:13]: can you please suggest how would it perform for an existing object?

**Pull request merged**